Aerolyzer/app/views.py

```diff
@@ -59,16 +62,87 @@ def gallery(request):
 
 @login_required
 def upload(request):
-    return render(request,
-    'app/upload.html',
-	{ 'user': request.user },
-    )
+	if request.method == 'POST' and request.FILES['myfile']:
+		myfile = request.FILES['myfile']
+		fs = FileSystemStorage()
+		filename = fs.save(myfile.name, myfile)
+		uploadedFileUrl = fs.url(filename)
+		request.session['filename'] = filename
+		request.session['uploadedFileUrl'] = uploadedFileUrl
+		# verified = imgRestFuncs(filename)
+		# if not verified['isVerified']:
+		# 	return render(request,
+		# 	'app/upload.html',
+		# 	{ 'user': request.user, 'filename': filename, 'uploadedFileUrl': uploadedFileUrl,
+		# 	'error_message': verified['data']},)
+		# else:
+		#	request.session['exifData'] = verified['data']
+		return HttpResponseRedirect('retrieve')
+	return render(request, 'app/upload.html', { 'user': request.user, },)
 
 @login_required
 def profile(request):
     return render(request,
     'app/profile.html',
-	{ 'user': request.user },
+	{ 'user': request.user, },
+    )
+
+@login_required
+def retrieve(request):
+	uploadedFileUrl = request.session['uploadedFileUrl']
+	filename = request.session['filename']
+	# exifData = request.session['exifData']
+
+	if request.method == 'POST':
+		# wunderData = retrieve_weather_info(exifData['location'])
```

Comment:
Just to be sure, the GPS fields are:
GPS Altitude
GPS Longitude
GPS Latitude
correct?